public/app/features/api-keys/ApiKeysPage.tsx

```diff
@@ -146,10 +170,85 @@ export class ApiKeysPageUnconnected extends PureComponent<Props, State> {
             ) : null}
           </>
         </Page.Contents>
+        <MigrationSummary
+          visible={this.state.operationSummaryVisible}
+          data={this.state.operationSummaryData}
+          dismissModal={this.dismissModal}
+        />
       </Page>
     );
   }
 }
+interface MigrationResult {
+  Total: number;
+  Migrated: number;
+  Failed: number;
+  FailedApikeyIDs: number[];
+  FailedDetails: string[];
+}
+
+export type MigrationSummaryProps = {
+  visible: boolean;
+  data: MigrationResult;
+  dismissModal: () => void;
+};
+
+const styles: { [key: string]: React.CSSProperties } = {
+  migrationSummary: {
+    padding: '20px',
+  },
+  infoText: {
+    color: '#007bff',
+  },
+  summaryDetails: {
+    marginTop: '20px',
+  },
+  summaryParagraph: {
+    margin: '10px 0',
+  },
+};
+
+export const MigrationSummary: React.FC<MigrationSummaryProps> = ({ visible, data, dismissModal }) => {
+  if (!visible) {
+    return null;
+  }
+
+  return (
+    <Modal title="Migration summary" isOpen={true} closeOnBackdropClick={true} onDismiss={dismissModal}>
+      <div style={styles.migrationSummary}>
+        {data.FailedApikeyIDs.length !== 0 && (
+          <p>Do not worry if you see failed API key migrations. All your API keys are functional and operational.</p>
```

Comment:
Sounds a bit strange. If _All your API keys are functional and operational_, should you perform any actions? But migration failed, so you probably should. Maybe show action tip only if there're keys failed to migrate?

Reply:
yea, it only shows if there are apikeys that failed. see line 219

Reply:
Can we elaborate a bit on error message? Maybe something more clear from an action perspective?

Reply:
yea for sure

Reply:
made the wording better

Reply:
@alexanderzobnin talked w. vesna and made the wording better and added a close button, as it is best practice to have a button for the modal